### PR TITLE
CI: Fail Numpy Dev build on DeprecationWarnings from numpy only

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

I believe the intent of `-W error` was to catch DeprecationWarnings from numpy primarily. Currently builds are failing because of a warning from distutils which idk if we can do anything about.

### CODE REVIEW

File: .github/workflows/posix.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
             env_file: actions-310-numpydev.yaml
             pattern: "not slow and not network and not single_cpu"
             pandas_testing_mode: "deprecate"
-            test_args: "-W error"
+            test_args: "-W error::DeprecationWarning:numpy"
```
Comment (jbrockmendel) [inline .github/workflows/posix.yml:64]: will this still catch FutureWarnings that we issue?

Comment (mroeschke) [reply .github/workflows/posix.yml:64]: Nope, only the DeprecationWarnings from numpy. Do we want to fail the build on our FutureWarnings?

Comment (jreback) [reply .github/workflows/posix.yml:64]: its possible we want to (but not this PR)

**Comment by jreback** [comment]
backport?

**Comment by mroeschke** [comment]
> backport?

Yup probably a good idea

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**